client/components/community-translator/index.jsx

```diff
@@ -158,8 +171,13 @@ class CommunityTranslator extends Component {
 	}
 
 	render() {
-		return null;
+		return <QueryUserSettings />;
 	}
 }
 
-export default localize( CommunityTranslator );
+const mapState = state => ( {
+	username: getCurrentUserName( state ),
+	translatorEnabled: getUserSetting( state, ENABLE_TRANSLATOR_KEY ),
+} );
+
+export default connect( mapState )( localize( CommunityTranslator ) );
```

Comment:
Two nits: the component doesn't seem to use the `localize` HOC, and using `PureComponent` is not necessary here: all Redux-connected components are pure by default.

Reply:
> the component doesn't seem to use the localize HOC,

I made a comment about this in the component, which I've moved down to the localize() call itself.

Even though we don't use `translate()` we still need this component to update when the translations change. In particular, we need to redraw to pick up changes when i18n finishes loading new data.

This has made the rerendering logic pretty complicated overall, and I've got to where it's at now by incrementally fixing failing updates. I think it's right now in the sense that it works, but I wonder if there's a conceptually better approach that I'm not seeing.

> all Redux-connected components are pure by default.

Thanks, :TIL: https://react-redux.js.org/api/connect#pure-boolean